src/chainstate/stacks/index/marf.rs

```diff
@@ -175,12 +179,13 @@ impl MARF {
     /// s must point to the target Trie
     fn root_copy(storage: &mut TrieFileStorage, prev_block_hash: &BlockHeaderHash) -> Result<(), Error> {
         let cur_block_hash = storage.get_cur_block();
-        storage.open_block(prev_block_hash, false)?;
+        storage.open_block(prev_block_hash)?;
+        let prev_block_identifier = storage.get_cur_block_identifier()?;
         
         let (mut prev_root, _) = Trie::read_root(storage)?;
-        let new_root_hash = MARF::node_copy_update(&mut prev_root, 1)?;
+        let new_root_hash = MARF::node_copy_update(&mut prev_root, prev_block_identifier)?;
         
-        storage.open_block(&cur_block_hash, true)?;
+        storage.open_block(&cur_block_hash)?;
```

Comment:
Hmm.  While it is the case that on-disk Tries are never written to, the in-RAM Trie is definitely written to.  I feel like the read/write vs read-only guard should remain in this method, in order to protect the in-RAM Trie from being accidentally corrupted (it would be better or the code to panic than write to a Trie when we expect it to be read-only).  Thoughts?

Reply:
I'd slightly prefer keeping the guard out. While it's true that the check _could_ guard writes to the TrieRAM instance, I don't think the check adds much if anything in this case: all of the code paths where `write_nodetype` is called directly by the MARF for copy-on-writes, the caller opens the block in write mode _immediately_ before calling write_nodetype, meaning the test would never fail in those cases. The other case where this check could enforce something is on leaf inserts. However, here too, the TrieRAM can only ever have been opened in write mode (because it's opened via a call to `extend_trie`). If it's not opened, it already throws an `WriteNotBegunError`

Reply:
Alright, sounds good.